presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestCustomJsonDateTimeFormatter.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json;
+
+import io.prestosql.plugin.kafka.encoder.json.format.FormatType;
+import io.prestosql.spi.type.SqlDate;
+import io.prestosql.spi.type.SqlTime;
+import io.prestosql.spi.type.SqlTimeWithTimeZone;
+import io.prestosql.spi.type.SqlTimestamp;
+import io.prestosql.spi.type.SqlTimestampWithTimeZone;
+import io.prestosql.spi.type.TimeZoneKey;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.testng.annotations.Test;
+
+import java.util.Locale;
+import java.util.Optional;
+
+import static io.prestosql.spi.type.TimeZoneKey.UTC_KEY;
+import static io.prestosql.spi.type.TimeZoneKey.getTimeZoneKey;
+import static io.prestosql.spi.type.Timestamps.PICOSECONDS_PER_MILLISECOND;
+import static io.prestosql.testing.assertions.Assert.assertEquals;
+import static java.util.concurrent.TimeUnit.DAYS;
+
+public class TestCustomJsonDateTimeFormatter
+{
+    private DateTimeFormatter getFormatterForPattern(String pattern)
+    {
+        return DateTimeFormat.forPattern(pattern).withLocale(Locale.ENGLISH);
+    }
+
+    private void testDate(SqlDate value, FormatType formatType, String formatHint)
+    {
+        String formattedStr = formatType.getFormatter(Optional.of(formatHint))
+                .formatDate(value);
```

Comment:
This could all be on one line.  The pattern shows up in many places, though, so it may not worth making all those changes.